### PR TITLE
Re add valhalla

### PR DESCRIPTION
**Pull request by chrstnbwnkl**
Fixes https://github.com/openstreetmap/openstreetmap-website/issues/3843

Renamed the FOSSGIS routing engine and the `fossgis.js` file, and added the Valhalla routing  engine (based off previous work back from the mapzen days).

### CODE REVIEW

**Comment by systemed** [comment]
Had a play around with this and it looks great.

One suggestion: the Graphhopper and OSRM instructions write out an index (in bold) at the start of each step. So:

> **3.** Turn right onto **Enstone Road**	90m

rather than just

> Turn right onto **Enstone Road**	90m

For consistency and for usability it would be good to include this in the Valhalla instructions. (It's particularly useful when discussing a shared route, e.g. "why is Valhalla sending me along Frog Street at step 7 on this route".)

Otherwise looks good to go.

File: app/assets/javascripts/index/directions/valhalla.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+          callback(true);
+        }
+      });
+    }
+  };
+}
+
+  OSM.Directions.addEngine(new ValhallaEngine("valhalla_car", "auto"), true);
+  OSM.Directions.addEngine(new ValhallaEngine("valhalla_bicycle", "bicycle"), true);
+  OSM.Directions.addEngine(new ValhallaEngine("valhalla_foot", "pedestrian"), true);
```
Comment (tomhughes) [inline app/assets/javascripts/index/directions/valhalla.js:112]: Can we add a `fossgis_` prefix to the engine names here please?

**Review by tomhughes — COMMENTED** [review]
My comments are all cosmetic, around making naming consistent.

I'm open to arguments that we shouldn't have the prefix, but we should be consistent in either having it or not having it - whether to have it really comes down to whether we're likely to have multiple instances of OSRM or Valhalla at any point...

**Comment by chrstnbwnkl** [comment]
Thanks for the comments, I added the manoeuvre numbering and renamed the routers!

**Comment by nilsnolde** [comment]
fwiw, I just realized the OSM homepage might be impacted a bit as well: the fossgis valhalla instance is down for a few days, see https://github.com/valhalla/valhalla/issues/6067#issuecomment-4480589599. we had repeated hardware failures and are considering options.

**Pull request merged**